frame/bounties/src/lib.rs

```diff
@@ -254,38 +223,64 @@ decl_error! {
 		/// The bounties cannot be claimed/closed because it's still in the countdown period.
 		Premature,
 	}
-}
-
-decl_module! {
-	pub struct Module<T: Config>
-		for enum Call
-		where origin: T::Origin
-	{
-		/// The amount held on deposit per byte within bounty description.
-		const DataDepositPerByte: BalanceOf<T> = T::DataDepositPerByte::get();
-
-		/// The amount held on deposit for placing a bounty proposal.
-		const BountyDepositBase: BalanceOf<T> = T::BountyDepositBase::get();
-
-		/// The delay period for which a bounty beneficiary need to wait before claim the payout.
-		const BountyDepositPayoutDelay: T::BlockNumber = T::BountyDepositPayoutDelay::get();
-
-		/// Bounty duration in blocks.
-		const BountyUpdatePeriod: T::BlockNumber = T::BountyUpdatePeriod::get();
-
-		/// Percentage of the curator fee that will be reserved upfront as deposit for bounty curator.
-		const BountyCuratorDeposit: Permill = T::BountyCuratorDeposit::get();
-
-		/// Minimum value for a bounty.
-		const BountyValueMinimum: BalanceOf<T> = T::BountyValueMinimum::get();
 
-		/// Maximum acceptable reason length.
-		const MaximumReasonLength: u32 = T::MaximumReasonLength::get();
-
-		type Error = Error<T>;
-
-		fn deposit_event() = default;
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	#[pallet::metadata(T::AccountId = "AccountId", BalanceOf<T> = "Balance")]
+	pub enum Event<T: Config> {
+		/// New bounty proposal. \[index\]
+		BountyProposed(BountyIndex),
+		/// A bounty proposal was rejected; funds were slashed. \[index, bond\]
+		BountyRejected(BountyIndex, BalanceOf<T>),
+		/// A bounty proposal is funded and became active. \[index\]
+		BountyBecameActive(BountyIndex),
+		/// A bounty is awarded to a beneficiary. \[index, beneficiary\]
+		BountyAwarded(BountyIndex, T::AccountId),
+		/// A bounty is claimed by beneficiary. \[index, payout, beneficiary\]
+		BountyClaimed(BountyIndex, BalanceOf<T>, T::AccountId),
+		/// A bounty is cancelled. \[index\]
+		BountyCanceled(BountyIndex),
+		/// A bounty expiry is extended. \[index\]
+		BountyExtended(BountyIndex),
+	}
 
+	// Note :: For backward compatibility reasons,
+	// pallet-bounties uses Treasury for storage.
+	// This is temporary solution, soon will get replaced with
+	// Own storage identifier.
```

Comment:
Hmmm, should be replaced with its own Storage identifier now? Not sure what that would break

Reply:
This is something that I'm actually not sure about, @shawntabrizi or @thiolliere, would there be any problems changing the pallet prefix from `Treasury` to `Bounties`?

Reply:
No it should be done one time, we can do it now.
But this message should be removed as it no longer use `Treasury` prefix anymore. And a migration function should be written to move the storages from the `Treasury` prefix to the pallet name